File: Medium/p_19.c
struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
    if(head==NULL){
        return head;
    }
    struct ListNode *temp=head,*point,*back;
    int count=0,position,i;
    while(temp!=NULL){
        count++;
        temp=temp->next;
    }
    temp=head;
    position=count-n+1;
    if(position==1){
        head=head->next;
        free(temp);
        return head;
    }
    for(i=2;i<position;i++){
        temp=temp->next;
    }
    point=temp->next;
    back=point->next;
    temp->next=back;
    free(point);
    return head;
}

File: Medium/p_19.java
class Solution {
    private int num=0;
    private ListNode back=null;
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null) return head;
        else if(head.next==null&&n==1) return null;
        remove(head,n);
        if(this.num==n) return head.next;
        return head;
    }
    private void remove(ListNode head,int n){
        if(head==null) return;
        remove(head.next,n);
        (this.num)++;
        if(this.num==(n-1)) back=head;
        else if(this.num==(n+1)){
            head.next=back;
            return;
        }
        return;
    }
}
